pipestat/backend.py

```diff
@@ -0,0 +1,1226 @@
+        self._data[self.pipeline_name].setdefault("sample", {})
+        with self._data as data_locked:
+            data_locked.write()
+
+    def _load_results_file(self) -> None:
+        _LOGGER.debug(f"Reading data from '{self.results_file_path}'")
+        data = YAMLConfigManager(filepath=self.results_file_path)
+        if not bool(data):
+            self._data = data
+            self._data.setdefault(self.pipeline_name, {})
+            self._data[self.pipeline_name].setdefault("project", {})
+            self._data[self.pipeline_name].setdefault("sample", {})
+            with self._data as data_locked:
+                data_locked.write()
+        namespaces_reported = [k for k in data.keys() if not k.startswith("_")]
+        num_namespaces = len(namespaces_reported)
+        if num_namespaces == 0:
+            self._data = data
+        elif num_namespaces == 1:
+            previous = namespaces_reported[0]
+            if self.pipeline_name != previous:
+                msg = f"'{self.results_file_path}' is already used to report results for a different (not {self.pipeline_name}) namespace: {previous}"
+                raise PipestatError(msg)
+            self._data = data
+        else:
+            raise PipestatError(
+                f"'{self.results_file_path}' is in use for {num_namespaces} namespaces: {', '.join(namespaces_reported)}"
+            )
+
+    def report(
+        self,
+        values: Dict[str, Any],
+        record_identifier: str,
+        pipeline_type: Optional[str] = None,
+        force_overwrite: bool = False,
+        # strict_type: bool = True,
+    ) -> None:
+        """
+        Update the value of a result in a current namespace.
+
+        This method overwrites any existing data and creates the required
+         hierarchical mapping structure if needed.
+
+        :param Dict[str, Any] values: dict of results identifiers and values
+            to be reported
+        :param str record_identifier: unique identifier of the record
+        :param str pipeline_type: "sample" or "project"
+        :param bool force_overwrite: Toggles force overwriting results, defaults to False
+        """
+
+        pipeline_type = pipeline_type or self.pipeline_type
+        record_identifier = record_identifier or self.record_identifier
+
+        result_identifiers = list(values.keys())
+        self.assert_results_defined(results=result_identifiers, pipeline_type=pipeline_type)
+        existing = self.list_results(
+            record_identifier=record_identifier,
+            restrict_to=result_identifiers,
+            pipeline_type=pipeline_type,
+        )
+        if existing:
+            existing_str = ", ".join(existing)
+            _LOGGER.warning(f"These results exist for '{record_identifier}': {existing_str}")
+            if not force_overwrite:
+                return False
+            _LOGGER.info(f"Overwriting existing results: {existing_str}")
+        # for r in result_identifiers:
```

Comment:
if you are not using this code, just clean it

Reply:
I cleaned up most of this. Leaving `r_id = self._strict_record_id(record_identifier)` for now until I'm sure the original functionality is captured.